torch/csrc/api/src/nn/modules/embedding.cpp

```diff
@@ -11,27 +11,79 @@
 namespace torch {
 namespace nn {
 
-EmbeddingOptions::EmbeddingOptions(int64_t count, int64_t dimension)
-    : count_(count), dimension_(dimension) {}
+EmbeddingOptions::EmbeddingOptions(int64_t num_embeddings, int64_t embedding_dim) : num_embeddings(num_embeddings), embedding_dim(embedding_dim) {}
 
-EmbeddingImpl::EmbeddingImpl(EmbeddingOptions options) : options(options) {
-  reset();
+EmbeddingImpl::EmbeddingImpl(int64_t num_embeddings, int64_t embedding_dim, c10::optional<int64_t> padding_idx, c10::optional<float> max_norm,
+float norm_type, bool scale_grad_by_freq, bool sparse, c10::optional<torch::Tensor> weight): EmbeddingImpl(EmbeddingOptions(num_embeddings, embedding_dim)){
+  if (padding_idx != c10::nullopt){
+    if(*padding_idx > 0){
+      assert((padding_idx < num_embeddings) && "Padding_idx must be within num_embeddings");
+    }
+    else{
+      assert((padding_idx >= -num_embeddings) && "Padding_idx must be within num_embedding");
+      *padding_idx = *padding_idx+num_embeddings;
+      options.padding_idx_ = padding_idx;
+    }
+  }
+  if(max_norm != c10::nullopt){
+    options.max_norm(max_norm);
+  }
+  options.norm_type(norm_type);
+  options.scale_grad_by_freq(scale_grad_by_freq);
+
+  if (!weight.has_value()){
+  // if (weight ==  c10::nullopt){
+    options.weight(torch::empty({num_embeddings, embedding_dim}));
+    EmbeddingImpl::reset();
+  }
+  else{
+    assert((((*weight).size(0) == num_embeddings) && ((*weight).size(1) == embedding_dim)) && "Shape of weight does not match num_embeddings and embedding_dim");
+  }
 }
 
 void EmbeddingImpl::reset() {
-  weight = register_parameter(
-      "weight", torch::empty({options.count_, options.dimension_}));
-  NoGradGuard guard;
-  weight.normal_(0, 1);
+  (*(options.weight_)).Tensor::normal_(0, 1);
```

Comment:
We can use `torch::nn::init::normal_(weight)`, if we include the `torch/nn/init.h` header.